Verizon/src/main/java/com/mopub/mobileads/VerizonRewardedVideo.java

```
@@ -0,0 +1,327 @@
+    }
+
+
+    protected String getSiteIdKey() {
+
+        return SITE_ID_KEY;
+    }
+
+
+    @Override
+    protected boolean checkAndInitializeSdk(@NonNull final Activity launcherActivity,
+        @NonNull final Map<String, Object> localExtras, @NonNull final Map<String, String> serverExtras) {
+
+        if (serverExtras == null || serverExtras.isEmpty()) {
+            MoPubLog.log(CUSTOM, ADAPTER_NAME, "Ad request to Verizon failed because " +
+                "serverExtras is null or empty");
+
+            return false;
+        }
+
+        // Cache serverExtras so siteId can be used to initalizate VAS early at next launch
+        verizonAdapterConfiguration.setCachedInitializationParameters(launcherActivity, serverExtras);
+
+        String siteId = serverExtras.get(getSiteIdKey());
+
+        if (!VASAds.isInitialized()) {
+            Application application = null;
+
+            if (launcherActivity != null) {
+                application = launcherActivity.getApplication();
+            }
+
+            if (application == null || !StandardEdition.initialize(application, siteId)) {
+                MoPubLog.log(LOAD_FAILED, ADAPTER_NAME, ADAPTER_CONFIGURATION_ERROR.getIntCode(), ADAPTER_CONFIGURATION_ERROR);
+
+                return false;
+            }
+        }
+
+        // The current activity must be set as resumed so VAS can track ad visibility
+        ActivityStateManager activityStateManager = VASAds.getActivityStateManager();
+        if (activityStateManager != null) {
+            activityStateManager.setState(launcherActivity, ActivityStateManager.ActivityState.RESUMED);
+        }
+
+        return true;
+    }
+
+
+    @Override
+    protected void loadWithSdkInitialized(@NonNull final Activity activity, @NonNull final Map<String, Object> localExtras,
+        @NonNull final Map<String, String> serverExtras) {
+
+        this.activity = activity;
+        placementId = serverExtras.get(getPlacementIdKey());
```

Get the placement ID in `checkAndInitializeSdk()` and fail fast in there if the placement ID is null or empty (instead of waiting to do that here).